common/meson.build

```diff
@@ -1,8 +1,23 @@
+git_dir = join_paths(source_root, '.git')
+git_args = ['git', '--git-dir=' + git_dir,
+            '--work-tree=' + source_root]
+
+is_git = run_command([git_args, 'rev-parse'], check: false)
+has_tag = run_command([git_args], 'describe', '--tags', check: false)
+
+if is_git.returncode() == 0 and has_tag.returncode() != 0
+    # put the mpv version and hash together
+    mpv_version = 'v' + meson.project_version().replace('-UNKNOWN', '')
+    hash = run_command([git_args, 'rev-parse', '--short', '@'],
```

Comment:
this will only work if meson configure is run.

Reply:
Well yeah the hash doesn't update with new commits if you're doing development on a shallow clone but nobody does that right? Could write a python script I guess.

Reply:
It's less than 10 lines as an inline script so not too bad.

Reply:
It might be all in python to avoid mixing meson/python logic.